### ISSUE TITLE
MultiConnect Timestamp (NaiveDateTime, PrimitiveDateTime, OffsetDateTime) not working.

### ISSUE THREAD

**Issue by BlackDex** [diesel-rs/diesel#3748]
## Setup

Simple small program using the MultiConnect derive with the following schema and table.
```rust
pub mod schema {
    table! {
        posts (id) {
            id -> Integer,
            title -> Varchar,
            body -> Text,
            updated -> Timestamp,
        }
    }
}

pub mod models {
    use chrono::NaiveDateTime;

    use crate::schema::posts;
    #[derive(Queryable, Insertable)]
    pub struct Post {
        pub id: i32,
        pub title: String,
        pub body: String,
        pub updated: NaiveDateTime,
    }
}
```

### Versions

- **Rust:** rustc 1.73.0-nightly (180dffba1 2023-08-14)
- **Diesel:** master branch (c99ebda62395a092e059d6ca494861afa559028b)
- **Database:** MultiConnection (sqlite, postgresql, mysql)
- **Operating System:** Linux

### Feature Flags

- **diesel:** "chrono", "r2d2"

## Problem Description
When trying to use `Timestamp` via `NaiveDateTime` using the `MultiConnection` it reports there is no `ToSql`, and probably it will also complain there is no `FromSql`, but i haven't gotten that far yet.

### What are you trying to accomplish?
Using the `MultiConnection` derive feature using a single schema/table layout.

### What is the expected output?
No errors, and a way to be working with at least `NaiveDateTime`.

### What is the actual output?
Compile error.

```
error[E0277]: the trait bound `NaiveDateTime: ToSql<diesel::sql_types::Timestamp, multi_connection_impl::backend::MultiBackend>` is not satisfied
    --> src/main.rs:245:26
     |
245  |                 .execute(conn).expect("Error during inserting user");
     |                  ------- ^^^^ the trait `ToSql<diesel::sql_types::Timestamp, multi_connection_impl::backend::MultiBackend>` is not implemented for `NaiveDateTime`
     |                  |
     |                  required by a bound introduced by this call
     |
     = help: the following other types implement trait `ToSql<A, DB>`:
               <NaiveDateTime as ToSql<diesel::sql_types::Timestamp, Mysql>>
               <NaiveDateTime as ToSql<diesel::sql_types::Timestamp, Pg>>
               <NaiveDateTime as ToSql<diesel::sql_types::Timestamp, Sqlite>>
               <NaiveDateTime as ToSql<diesel::sql_types::Nullable<diesel::sql_types::Timestamp>, __DB>>
               <NaiveDateTime as ToSql<diesel::sql_types::Nullable<Datetime>, __DB>>
               <NaiveDateTime as ToSql<diesel::sql_types::Nullable<diesel::sql_types::Timestamptz>, __DB>>
               <NaiveDateTime as ToSql<Datetime, Mysql>>
               <NaiveDateTime as ToSql<diesel::sql_types::Timestamptz, Pg>>
               <NaiveDateTime as ToSql<TimestamptzSqlite, Sqlite>>
     = note: required for `diesel::expression::bound::Bound<diesel::sql_types::Timestamp, NaiveDateTime>` to implement `QueryFragment<multi_connection_impl::backend::MultiBackend>`
     = note: 1 redundant requirement hidden
     = note: required for `ColumnInsertValue<schema::posts::columns::updated, diesel::expression::bound::Bound<diesel::sql_types::Timestamp, NaiveDateTime>>` to implement `QueryFragment<multi_connection_impl::backend::MultiBackend>`
     = note: required for `ColumnInsertValue<schema::posts::columns::updated, diesel::expression::bound::Bound<diesel::sql_types::Timestamp, NaiveDateTime>>` to implement `InsertValues<schema::posts::table, multi_connection_impl::backend::MultiBackend>`
     = note: 3 redundant requirements hidden
     = note: required for `InsertStatement<table, ValuesClause<(ColumnInsertValue<title, Bound<Text, String>>, ColumnInsertValue<body, Bound<Text, String>>, ColumnInsertValue<published, Bound<Bool, bool>>, ColumnInsertValue<updated, Bound<Timestamp, NaiveDateTime>>), table>>` to implement `QueryFragment<multi_connection_impl::backend::MultiBackend>`
     = note: the full type name has been written to '/home/mathijs/rust/dieselmulticrate/target/debug/deps/dieselmulticrate-ddcbaeaecb4c0ddd.long-type-901884936185368146.txt'
     = note: required for `InsertStatement<table, ValuesClause<(ColumnInsertValue<title, Bound<Text, String>>, ColumnInsertValue<body, Bound<Text, String>>, ColumnInsertValue<published, Bound<Bool, bool>>, ColumnInsertValue<updated, Bound<Timestamp, NaiveDateTime>>), table>>` to implement `ExecuteDsl<InferConnection, multi_connection_impl::backend::MultiBackend>`
     = note: the full type name has been written to '/home/mathijs/rust/dieselmulticrate/target/debug/deps/dieselmulticrate-ddcbaeaecb4c0ddd.long-type-901884936185368146.txt'
note: required by a bound in `diesel::RunQueryDsl::execute`
    --> /home/mathijs/rust/diesel/diesel/src/query_dsl/mod.rs:1432:15
     |
1429 |     fn execute(self, conn: &mut Conn) -> QueryResult<usize>
     |        ------- required by a bound in this associated function
...
1432 |         Self: methods::ExecuteDsl<Conn>,
     | 
```

### Are you seeing any additional errors?
No (not  yet)

### Steps to reproduce

Use the schema/table layout from above.
And this is a create table for Postgresql as an example.
```sql
CREATE TABLE posts (
  id            INTEGER   NOT NULL PRIMARY KEY,
  title         TEXT      NOT NULL,
  body          TEXT      NOT NULL,
  updated       TIMESTAMP NOT NULL
);
```

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) and the [discussion forum](https://github.com/diesel-rs/diesel/discussions) for similar possible closed issues.
 
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by BlackDex** [comment]
Ok, i did some checking, and it basically is the same issues as it was with #3747.
It is just missing the correct from/to sql impl's generated via `generate_bind_collector`.

The issue i have is that I'm somehow unable to conditionally generate them if chrono is enabled or not.
Adding `chrono = []` to `diesel_derives` and `chrono = ["diesel_derives/chrono", "dep:chrono"]` doesn't seem to pass this feature on. Or something with `quote!` is not handling it for some reason.

If i just ignore the features and just add `use chrono::NaiveDateTime` to `mod multi_connection_impl` and add `(quote::quote!(diesel::sql_types::Timestamp), quote::quote!(NaiveDateTime))` to both the `to_sql` and `from_sql`, it seems to work for me.

Hopefully this makes fixing the issue a bit more easier :)

**Comment by weiznich** [comment]
Closed by #3752
